src/unix/haiku/mod.rs

```diff
@@ -0,0 +1,729 @@
+use dox::mem;
+
+#[cfg(target_pointer_width = "32")]
+pub type c_long = i32;
+#[cfg(target_pointer_width = "64")]
+pub type c_long = i64;
+#[cfg(target_pointer_width = "32")]
+pub type c_ulong = u32;
+#[cfg(target_pointer_width = "64")]
```

Comment:
Can these `#[cfg]` get pushed down farther instead of being here? (e.g. a b32 and b64 module)